src/language.rs

```diff
@@ -0,0 +1,54 @@
+use clap::ValueEnum;
+use tree_sitter::Language;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, ValueEnum)]
+pub enum SupportedLanguageName {
+    Rust,
+    Typescript,
+}
+
+impl SupportedLanguageName {
+    pub fn get_language(self) -> Box<dyn SupportedLanguage> {
+        match self {
+            Self::Rust => Box::new(get_rust_language()),
+            Self::Typescript => Box::new(get_typescript_language()),
+        }
+    }
+}
+
+pub trait SupportedLanguage {
+    fn language(&self) -> Language;
+    fn name_for_ignore_select(&self) -> &'static str;
+}
+
+pub struct SupportedLanguageRust;
+
+impl SupportedLanguage for SupportedLanguageRust {
+    fn language(&self) -> Language {
+        tree_sitter_rust::language()
+    }
+
+    fn name_for_ignore_select(&self) -> &'static str {
+        "rust"
+    }
+}
+
+pub fn get_rust_language() -> SupportedLanguageRust {
+    SupportedLanguageRust
+}
+
+pub struct SupportedLanguageTypescript;
+
+impl SupportedLanguage for SupportedLanguageTypescript {
+    fn language(&self) -> Language {
+        tree_sitter_typescript::language_tsx()
```

Comment:
Looks like tsx vs typescript are two different tree-sitter grammars, ultimately I'd guess the nicest ergonomics would be to eg try and parse the given query as both (and invoke the appropriate per-grammar parsed query based on each source file's extension?) and if it can't parse it for whichever grammar then silently skip those files?